packages/backend/src/util/handlers/createExpressSynthqlHandler.ts

```diff
@@ -0,0 +1,31 @@
+import { QueryEngine } from '../..';
+import { Request, Response } from 'express';
+
+type ExpressSynthqlHandler = (req: Request, res: Response) => void;
+
+export async function createExpressSynthqlHandler<T>(
+    queryEngine: QueryEngine<T>,
+): Promise<ExpressSynthqlHandler> {
+    return async (req, res) => {
+        const headers = req.headers;
+
+        const query = JSON.parse(req.body);
+
+        res.status(200);
+        res.setHeader('Content-Type', 'application/x-ndjson');
+
+        try {
+            for await (const intermediateResult of queryEngine.execute(query, {
+                returnLastOnly: headers['x-return-last-only'] === 'true',
+            })) {
+                res.write(JSON.stringify(intermediateResult) + '\n');
+            }
+
+            res.end('SynthQL request handled successfully!');
+        } catch (error) {
+            res.write(JSON.stringify({ error: String(error) }) + '\n');
+
+            res.end('SynthQL request failed with error!');
```

Comment:
What's the reasoning for this?

Reply:
If I'm not being mistaken this is incompatible with the JSONL format on the react client.

Reply:
The idea was to listen for `res.end` to be called with that value, but you're right that it's not consistent with the `nl-json` format. Will remove that now.

Reply:
Notice that the client expects every line to be JSON so this would have broken the client. This would have been caught by a test.